src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,219 @@
+                    + driverId, e);
+        }
+        for (Car car : cars) {
+            List<Driver> allDriversByCarId = getDriversByCarId(car.getId());
+            car.setDrivers(allDriversByCarId);
+        }
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarRequest
+                = "UPDATE cars SET model = ? , manufacturer_id = ? WHERE id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(updateCarRequest)) {
+            statement.setString(1,car.getModel());
+            statement.setLong(2,car.getManufacturer().getId());
+            statement.setLong(3,car.getId());
+            statement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update Car with id "
+                    + car.getId() + " into DB",e);
+        }
+        deleteDriversFromCar(car);
+        addDriversToCar(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarQuery
+                = "UPDATE cars SET is_deleted = true WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement =
+                        connection.prepareStatement(deleteCarQuery)) {
+            statement.setLong(1,id);
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete Car with id "
+                    + id + " into DB",e);
+        }
+    }
+
+    private boolean addDriversToCar(Car car) {
+        String query = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                statement.setLong(2, driver.getId());
+            }
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't add driver by id into DB ",e);
+        }
+    }
+
+    private boolean deleteDriversFromCar(Car car) {
+        String deleteDriversByCarIdRequest =
+                "DELETE FROM cars_drivers WHERE cars_drivers.car_id = ?;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(deleteDriversByCarIdRequest)) {
+            statement.setLong(1, car.getId());
+            return statement.executeUpdate() != 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete Drivers by Car's id in DB ",e);
+        }
+    }
```

Comment:
![image](https://user-images.githubusercontent.com/81781479/199963946-b9980cca-2d7a-4167-b350-d7bc67a1fbf7.png)
those methods should be only in service